chunks/lang-rust/Dockerfile

```diff
@@ -6,22 +6,22 @@ USER gitpod
 # Dazzle does not rebuild a layer until one of its lines are changed. Increase this counter to rebuild this layer.
 ENV TRIGGER_REBUILD=1
 
+ENV PATH=$PATH:$HOME/.cargo/bin
 RUN cp /home/gitpod/.profile /home/gitpod/.profile_orig && \
     curl -fsSL https://sh.rustup.rs | sh -s -- -y --profile minimal --default-toolchain 1.58.1 \
-    && .cargo/bin/rustup component add \
+    && rustup component add \
         rls \
         rust-analysis \
         rust-src \
         rustfmt \
-    && .cargo/bin/rustup completions bash | sudo tee /etc/bash_completion.d/rustup.bash-completion > /dev/null \
-    && .cargo/bin/rustup completions bash cargo | sudo tee /etc/bash_completion.d/rustup.cargo-bash-completion > /dev/null \
-    && grep -v -F -x -f /home/gitpod/.profile_orig /home/gitpod/.profile > /home/gitpod/.bashrc.d/80-rust
-ENV PATH=$HOME/.cargo/bin:$PATH
-# TODO: setting CARGO_HOME to /workspace/.cargo avoids manual updates. Remove after full workspace backups are GA.
+    && rustup completions bash | sudo tee /etc/bash_completion.d/rustup.bash-completion > /dev/null \
+    && rustup completions bash cargo | sudo tee /etc/bash_completion.d/rustup.cargo-bash-completion > /dev/null \
+    && printf '%s\n' "$(grep -v -F -x -f /home/gitpod/.profile_orig /home/gitpod/.profile)" \
+    'mkdir -m 0755 -p "${CARGO_HOME:-/workspace/.cargo}" 2>/dev/null' > $HOME/.bashrc.d/80-rust \
+    && mkdir -m 0755 -p $HOME/.config/fish \
+    && printf '%s\n' 'alias rustup="env -u CARGO_HOME (command -v rustup)"' >> $HOME/.config/fish/config.fish \
+    && printf '%s\n' 'alias rustup="env -u CARGO_HOME $(command -v rustup)"' >> $HOME/.bashrc.d/80-rust \
+    && printf '%s\n' 'alias rustup="env -u CARGO_HOME $(command -v rustup)"' >> $HOME/.zshrc
```

Comment:
I am not sure about this. Are we updating other shells in other languages? I doubt it. Let us stick to the pattern. You can refer to contribution guide to learn why we prefer creating a dedicated rc in rc.d rather than appending it to existing rc files.

Reply:
This is my proposed solution, please check the PR description. Re-posting a part of the desc here:

- About #464
> I came across three solutions
> 1. Wrapping `rustup` with a bash script to unset `CARGO_HOME` variable to make it use the original `$HOME/.cargo` dir. Might not be future proof.
> 2. Greedily symlinking everything from `$HOME/.cargo/bin` to `/workspace/.cargo/bin`(Custom `CARGO_HOME`). Might cause unexpected issues.
> 3. Using shell specific aliases to wrap `rustup` in a quite neat way, what I ended up with.

> @princerachit: You can refer to contribution guide to learn why we prefer creating a dedicated rc in rc.d rather than appending it to existing rc files.

Yeah I totally agree, I'm doing that for the `bash` one(`$HOME/.bashrc.d/80-rust`), but for zsh and fish we don't have any custom logic set in their main shellrc file for automatically loading dedicated rc.d files, which is why I'm directly appending into the main shellrc. That's something we need to do from the `base` image I suppose.

Reply:
>which is why I'm directly appending into the main shellrc. That's something we need to do from the base image I suppose.

I agree. That is why I would prefer not to have the shell changes as part of the PR. This will set a wrong expectation for future contributors. We need to figure out if we want to support other shells. Can you create an issue for it?

Reply:
Since we were not supporting other shells let us skip it for now. We can revisit it later if required

Reply:
I actually made the attempt to support other shells in this specific PR after finding out that the `base` image installs `zsh` and `fish` alongside `bash` 😅 
https://github.com/gitpod-io/workspace-images/blob/d43c719bb2ff7b6849c0456fe21e3ca06c20168f/base/Dockerfile#L30

Reply:
Will create an issue and let you know 🙏

Reply:
I went back to my proposed solution 1 instead of using shell aliases.